internal/cmd/ske/kubeconfig/login/login.go

```diff
@@ -0,0 +1,240 @@
+			}
+
+			// cert not expired, nor will it expire in the next 15min; therefore, use the cached kubeconfig
+			if err := output(cmd, model.CacheKey, cachedKubeconfig); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	return cmd
+}
+
+type inputModel struct {
+	ProjectId   string
+	ClusterName string
+	CacheKey    string
+}
+
+type SKEClusterConfig struct {
+	STACKITProjectID string `json:"stackitProjectId"`
+	ClusterName      string `json:"clusterName"`
+}
+
+func parseInput() (*inputModel, error) {
+	obj, _, err := exec.LoadExecCredentialFromEnv()
+	if err != nil {
+		return nil, fmt.Errorf("LoadExecCredentialFromEnv: %w", err)
+	}
+
+	if err := clientauthenticationv1.AddToScheme(scheme.Scheme); err != nil {
+		return nil, err
+	}
+
+	obj, err = scheme.Scheme.ConvertToVersion(obj, clientauthenticationv1.SchemeGroupVersion)
+	if err != nil {
+		return nil, fmt.Errorf("ConvertToVersion: %w", err)
+	}
+
+	execCredential, ok := obj.(*clientauthenticationv1.ExecCredential)
+	if !ok {
+		return nil, fmt.Errorf("conversion to ExecCredential failed")
+	}
+	if execCredential == nil || execCredential.Spec.Cluster == nil {
+		return nil, fmt.Errorf("ExecCredential contains not all needed fields")
+	}
+	config := &SKEClusterConfig{}
+	err = json.Unmarshal(execCredential.Spec.Cluster.Config.Raw, config)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal: %w", err)
+	}
+
+	return &inputModel{
+		ClusterName: config.ClusterName,
+		ProjectId:   config.STACKITProjectID,
+		CacheKey:    fmt.Sprintf("ske-login-%x", sha256.Sum256([]byte(execCredential.Spec.Cluster.Server))),
+	}, nil
+}
+
+func getCachedKubeConfig(key string) *rest.Config {
+	cachedKubeconfig, err := cache.GetObject(key)
+	if err != nil {
+		return nil
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(cachedKubeconfig)
+	if err != nil {
+		return nil
+	}
+
+	return restConfig
+}
+
+func GetAndOutputKubeconfig(ctx context.Context, cmd *cobra.Command, apiClient *ske.APIClient, model *inputModel, fallbackToCache bool, cachedKubeconfig *rest.Config) error {
+	req := buildRequest(ctx, apiClient, model)
+	kubeconfigResponse, err := req.Execute()
+	if err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("request kubeconfig: %w", err)
+	}
+
+	kubeconfig, err := clientcmd.RESTConfigFromKubeConfig([]byte(*kubeconfigResponse.Kubeconfig))
+	if err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("parse kubeconfig: %w", err)
+	}
+	if err = cache.PutObject(model.CacheKey, []byte(*kubeconfigResponse.Kubeconfig)); err != nil {
+		if fallbackToCache {
+			return output(cmd, model.CacheKey, cachedKubeconfig)
+		}
+		return fmt.Errorf("cache kubeconfig: %w", err)
+	}
+
+	return output(cmd, model.CacheKey, kubeconfig)
+}
+
+func buildRequest(ctx context.Context, apiClient *ske.APIClient, model *inputModel) ske.ApiCreateKubeconfigRequest {
```

Comment:
consider adding unit test for this function

Reply:
The function contains no logic only assignments

Reply:
We have a unit test for this `buildRequest` function in all CLI commands, it's quite easy to write and in this case can test that the request includes a payload with the `expirationSeconds`.

For uniformity with the rest of the codebase, it should be added